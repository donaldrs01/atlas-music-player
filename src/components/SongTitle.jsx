import React from 'react';

const SongTitle = ({ title, artist }) => {
    return (
        <div className='text-left mt-4 w-full mb-6'>
            {/* Song Title */}
            <h1 className="text-xl font-bold">{title}</h1>
            {/* Artist */}
            <p className="text-sm text-gray-500 py-2">{artist}</p>
        </div>
    );
};

export default SongTitle;