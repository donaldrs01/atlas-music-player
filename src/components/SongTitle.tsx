import React from "react";

type SongTitleProps = {
  title: string;
  artist: string;
};

const SongTitle: React.FC<SongTitleProps> = ({ title, artist }) => {
  return (
    <div className="mt-6 w-full text-left">
      {/* Song Title */}
      <h1 className="text-light-highlight dark:text-dark-highlight text-4xl font-bold">
        {title}
      </h1>
      {/* Artist */}
      <p className="text-light-highlight dark:text-dark-highlight py-2 text-2xl">
        {artist}
      </p>
    </div>
  );
};

export default SongTitle;
