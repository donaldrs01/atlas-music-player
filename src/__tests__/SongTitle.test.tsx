import { render } from "@testing-library/react";
import SongTitle from "../components/SongTitle";
import { describe, test, expect } from "vitest";

describe("SongTitle Snapshot Tests", () => {
  test("renders a song title and artist", () => {
    const { container } = render(<SongTitle title="Schism" artist="Tool" />);
    expect(container).toMatchSnapshot();
  });

  test("renders a longer song title and artist", () => {
    const { container } = render(
      <SongTitle
        title="Carry on My Wayward Wayward Wayward Son"
        artist="NotKansas"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
