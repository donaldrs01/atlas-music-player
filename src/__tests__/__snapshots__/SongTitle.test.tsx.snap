// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Snapshot Tests > renders a longer song title and artist 1`] = `
<div>
  <div
    class="w-full text-left"
  >
    <h1
      class="ml-4 mt-4 text-2xl font-bold text-light-highlight dark:text-dark-highlight md:text-4xl"
    >
      Carry on My Wayward Wayward Wayward Son
    </h1>
    <p
      class="ml-4 text-xl text-light-highlight dark:text-dark-highlight md:py-2 md:text-2xl"
    >
      NotKansas
    </p>
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders a song title and artist 1`] = `
<div>
  <div
    class="w-full text-left"
  >
    <h1
      class="ml-4 mt-4 text-2xl font-bold text-light-highlight dark:text-dark-highlight md:text-4xl"
    >
      Schism
    </h1>
    <p
      class="ml-4 text-xl text-light-highlight dark:text-dark-highlight md:py-2 md:text-2xl"
    >
      Tool
    </p>
  </div>
</div>
`;
